src/components/MediaPlayer/VideoJSPlayer.js

```diff
@@ -116,66 +141,197 @@ function VideoJSPlayer({
         console.log('play');
         playerDispatch({ isPlaying: true, type: 'setPlayingStatus' });
       });
+      player.on('seeked', () => {
+        handleSeeked();
+      });
+      player.on('timeupdate', () => {
+        handleTimeUpdate();
+      });
     }
   }, [player]);
 
+  /**
+   * Switch canvas when using structure navigation / the media file ends
+   */
+  React.useEffect(() => {
+    if (isClicked && canvasIndex !== cIndex) {
+      switchPlayer();
+    }
+    setCIndex(canvasIndex);
+    setCanvasSegments(getSegmentMap({ manifest, canvasIndex }));
+  }, [canvasIndex]);
+
+  /**
+   * Update markers whenever player's currentTime is being
+   * updated. Time update happens when;
+   * 1. using structure navigation
+   * 2. seek and scrubbing events are fired
+   * 3. timeupdate event fired when playing the media file
+   */
   React.useEffect(() => {
     if (!player || !currentPlayer) {
       return;
     }
-
-    if (startTime != null) {
-      player.currentTime(startTime, playerDispatch({ type: 'resetClick' }));
-
-      // Mark current timefragment
+    if (currentNavItem != null && isReady) {
```

Comment:
Is there a reason to not use `currentNavItem !== null` here?   Just a heads up coercion could possibly bring up some odd behavior...maybe?